### PR TITLE
Character Manager

### PR DESCRIPTION
**Pull request by TomMalbran**
Moved the Character list and related functions from World into a new Character Manager and moved the Character clases to a new Character folder.

### CODE REVIEW

File: Assets/Scripts/Controllers/CharacterSpriteController.cs [hunk 1]
```diff
@@ -95,47 +95,46 @@ protected override void OnCreated(Character c)
...
+        GameObject char_go = objectGameObjectMap[character];
 
-        char_go.transform.position = new Vector3(c.X, c.Y, 0);
+        char_go.transform.position = new Vector3(character.X, character.Y, 0);
     }
 
-    protected override void OnRemoved(Character c)
+    protected override void OnRemoved(Character character)
     {
-        c.OnCharacterChanged -= OnChanged;
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/CharacterSpriteController.cs]: Why removing this? How are character.OnCharacterChanged callbacks getting cleaned up?

Comment (TomMalbran) [inline Assets/Scripts/Controllers/CharacterSpriteController.cs]: That is a mistake. I was trying to have a global Changed Event, but then decided not to, and missed to roll this back. Will fix it.

File: Assets/Scripts/Models/Character/CharacterManager.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        {
+            characters[i].Update(deltaTime);
+        }
+    }
+
+    /// <summary>
+    /// Create a Character in the specified tile.
+    /// </summary>
+    /// <param name="tile">The tile.</param>
+    public Character Create(Tile tile)
```
Comment (mikejbrown) [inline Assets/Scripts/Models/Character/CharacterManager.cs]: Stylecop `SA1632 : CSharp.Documentation : The documentation text within the param tag must be at least 10 characters in length. Documentation failing to meet this guideline most likely does not follow a proper grammatical structure required for documentation text. (Assembly-CSharp)`

File: Assets/Scripts/Models/Character/CharacterManager.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /// <summary>
+    /// Create a Character in the specified tile, with the specified color, uniform color and skin color.
+    /// </summary>
+    /// <param name="tile">The tile.</param>
+    /// <param name="color">The color.</param>
+    /// <param name="uniformColor">The  uniform color.</param>
+    /// <param name="skinColor">The  skin color.</param>
+    public Character Create(Tile tile, Color color, Color uniformColor, Color skinColor)
```
Comment (mikejbrown) [inline Assets/Scripts/Models/Character/CharacterManager.cs]: Stylecop `SA1632 : CSharp.Documentation : The documentation text within the param tag must be at least 10 characters in length. Documentation failing to meet this guideline most likely does not follow a proper grammatical structure required for documentation text. (Assembly-CSharp)`

File: Assets/Scripts/Controllers/CharacterSpriteController.cs (hunk 1 above)
Comment (mikejbrown) [inline Assets/Scripts/Controllers/CharacterSpriteController.cs]: :+1:

**Comment by mikejbrown** [comment]
This pull request has merge conflicts. Please resolve the conflicts and remove [Conflicts] from the title.

**Comment by TomMalbran** [comment]
Fixed StyleCop errors and Conflicts.

**Comment by mikejbrown** [comment]
Merged. By the way, "Re-add removed removal of changed" may not be the most helpful of commit messages but I still :heart: it. :grinning:

**Pull request merged**